client/packages/invoices/src/InboundShipment/ListView/AppBarButtons.tsx

```diff
@@ -95,35 +104,25 @@ export const AppBarButtons = ({
           label={t('button.export')}
         />
       </Grid>
-
-      {showManuallyLinkModal && (
-        <LinkInternalOrderModal
-          requestRequisitions={data?.nodes}
-          isOpen={linkRequestModalController.isOn}
-          onClose={linkRequestModalController.toggleOff}
-          onRowClick={row => {
-            createInvoice(name?.id ?? '', row.id);
-            linkRequestModalController.toggleOff();
-          }}
-          isLoading={internalOrderIsLoading}
-          onNextClick={() => {
-            if (name) {
-              createInvoice(name.id);
-            }
-          }}
-        />
-      )}
+      <LinkInternalOrderModal
+        isOpen={linkRequestModalController.isOn}
+        onClose={linkRequestModalController.toggleOff}
+        onRowClick={row => {
+          createInvoice(name?.id ?? '', row.id);
+          linkRequestModalController.toggleOff();
+        }}
+        onNextClick={() => {
+          if (name) {
+            createInvoice(name.id);
+          }
+        }}
+        name={name}
+      />
       <SupplierSearchModal
         open={invoiceModalController.isOn}
         onClose={invoiceModalController.toggleOff}
         onChange={async nameRow => {
-          setName(nameRow);
-          invoiceModalController.toggleOff();
-          if (manuallyLinkInternalOrder) {
-            linkRequestModalController.toggleOn();
-          } else {
-            createInvoice(nameRow.id);
-          }
+          handleRowClick(nameRow);
```

Comment:
Was already an issue, should fix while here... not returning here means the promise is not returned, if SupplierSearchModal was `await`ing the onChange, it wouldn't get any result.

Should either add `return`, or now that its just one line, changing to this would probably be best:

```tsx
    onChange={handleSupplierSelected}
```